llsd/serde_binary.py

```diff
@@ -232,9 +231,22 @@ def parse_binary(something):
     :param something: The data to parse in an indexable sequence.
     :returns: Returns a python object.
     """
-    if starts_with(b'<?llsd/binary?>', something):
-        just_binary = something.split(b'\n', 1)[1]
-    else:
-        just_binary = something
-    return LLSDBinaryParser().parse(just_binary)
+    # Try to match header, and if matched, skip past it.
+    remainder = match_binary_hdr(something)
+    # If we matched the header, then parse whatever follows, else parse the
+    # original bytes object or stream.
+    return parse_binary_noh(remainder if remainder is not None else something)
+
+
+def match_binary_hdr(something):
+    return matchseq(something, b'<? llsd/binary ?>')
 
+
+def parse_binary_noh(something):
```

Comment:
`"noh"` is cute but its meaning is not obvious. Could the header option be an optional kwarg of the simpler `parse_*` functions or could the name be made clearer?

Reply:
How about `parse_binary_nohdr()`? Or even `_parse_binary_nohdr()` since I view this as primarily internal?

I want to be able to pass control directly to this code from `llsd.parse()` when the generic parser recognizes a particular header, instead of passing and testing a bool.